src/engraving/tests/environment.cpp

```diff
@@ -51,14 +51,24 @@ static muse::testing::SuiteEnvironment engraving_se(
 
     mu::engraving::loadInstrumentTemplates(":/data/instruments.xml");
 
-    std::shared_ptr<::testing::NiceMock<mu::engraving::EngravingConfigurationMock> > configurator
-        = std::make_shared<::testing::NiceMock<mu::engraving::EngravingConfigurationMock> >();
+    using ECMock = ::testing::NiceMock<mu::engraving::EngravingConfigurationMock>;
+
+    std::shared_ptr<ECMock> configurator(new ECMock(), [](ECMock*) {}); // no delete
     ON_CALL(*configurator, isAccessibleEnabled()).WillByDefault(::testing::Return(false));
     ON_CALL(*configurator, defaultColor()).WillByDefault(::testing::Return(muse::draw::Color::BLACK));
-    mu::engraving::EngravingItem::engravingConfiguration.set(configurator);
+
+    muse::modularity::globalIoc()->unregister<mu::engraving::IEngravingConfiguration>("utests");
+    muse::modularity::globalIoc()->registerExport<mu::engraving::IEngravingConfiguration>("utests", configurator);
 },
 
     []() {
-    mu::engraving::EngravingItem::engravingConfiguration.set(nullptr);
+    std::shared_ptr<mu::engraving::IEngravingConfiguration> mock
+        = muse::modularity::globalIoc()->resolve<mu::engraving::IEngravingConfiguration>("utests");
+    muse::modularity::globalIoc()->unregister<mu::engraving::IEngravingConfiguration>("utests");
+
+    int use_count = mock.use_count();
+    LOGD() << "==~~~~~~~~~~~~~ use_count: " << use_count;
+    mu::engraving::IEngravingConfiguration* ecptr = mock.get();
+    delete ecptr;
```

Comment:
Could you explain a bit about what this does?

Reply:
The problem there is that there are still live pointers to the EngravingConfigurationMock, because of this the mock generates an error stating that it has not been deleted. 
This is a hack to delete it and get rid of its error. 
The problem itself is deeper, it seems that some scores objects are not deleted.

Added comment